airflow/operators/mysql_to_s3_operator.py

```diff
@@ -0,0 +1,117 @@
+    :type s3_bucket: str
+    :param s3_key: desired key for the file. It includes the name of the file.
+        If a csv file is wanted, the param must end with ".csv".
+    :type s3_key: str
+    :param mysql_conn_id: reference to a specific mysql database
+    :type mysql_conn_id: str
+     :param aws_conn_id: reference to a specific S3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - False: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - path/to/cert/bundle.pem: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    """
+
+    @apply_defaults
+    def __init__(
+            self,
+            query: str,
+            s3_bucket: str,
+            s3_key: str,
+            mysql_conn_id: str = 'mysql_default',
+            aws_conn_id: str = 'aws_default',
+            verify: Optional[Union[bool, str]] = None,
+            header: Optional[bool] = False,
+            index: Optional[bool] = False,
+            *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.query = query
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.mysql_conn_id = mysql_conn_id
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.header = header
+        self.index = index
+
+    def fill_na_with_none(self, series):
+        """Replace NaN values with None"""
+        return np.where(series.isnull(), None, series)
+
+    def fix_int_dtypes(self, df):
+        """
+        Mutate DataFrame to set dtypes for int columns containing NaN values."
+        """
+        for col in df:
+            if "float" in df[col].dtype.name and df[col].hasnans:
+                # inspect values to determine if dtype of non-null values is int or float
+                notna_series = df[col].dropna().values
+                if np.isclose(notna_series, notna_series.astype(int)).all():
+                    # set to dtype that retains integers and supports NaNs
+                    df[col] = self.fill_na_with_none(df[col]).astype(pd.Int64Dtype)
+
+    def execute(self, context, **kwargs):
+        self.hook = MySqlHook(mysql_conn_id=self.mysql_conn_id)
+        self.s3_conn = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+        data_df = self.hook.get_pandas_df(self.query)
+        self.log.info("Data from MySQL obtained")
+
+        self.fix_int_dtypes(data_df)
+        file_obj = StringIO()
+        data_df.to_csv(file_obj, header=self.header, index=self.index)
+        self.log.info("File_obj created")
+
```

Comment:
Some changes will be needed to conform the DataFrame into a file object that can be uploaded to S3. From the [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.upload_fileobj):
> At a minimum, it must implement the read method, and must return bytes.

Line 22 will need an update to:
```Python
from io import BytesIO, StringIO
```

```suggestion
file_obj.seek(0)
file_obj = BytesIO(bytes(file_obj.read(), "utf-8"))
```

Reply:
Thanks for your comment. In addition to that I have realized that I wrote load_file_obj function instead of upload_fileobj. Anyway, I was thinking of simplifying the code and not using string objects at all and take advantage of pandas s3 files handling ([https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.20.0.html#s3-file-handling,pandas doc](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.20.0.html#s3-file-handling))

Reply:
My opinion is to not include another dependency. But, based on  https://github.com/apache/airflow/pull/6670#discussion_r362535708 this section should likely be updated to use temp file instead.